src/HealthGPS.Tests/Channel.Test.cpp

```diff
@@ -234,7 +235,8 @@ TEST(ChannelTest, MultipleConsumers) {
     });
 
     producer.join();
-    std::for_each(threads.begin(), threads.end(), [](std::jthread &thread) { thread.join(); });
+    tbb::parallel_for_each(threads.begin(), threads.end(),
+                           [](std::jthread &thread) { thread.join(); });
```

Comment:
You don't want to run this in parallel. It could just be a ranged `for`-loop:

```suggestion
    for (auto &thread : threads) {
        thread.join();
    }
```